Mathlib/Topology/Algebra/LinearTopology.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández, Anatole Dedecker
+-/
+
+import Mathlib.RingTheory.TwoSidedIdeal.Operations
+import Mathlib.Topology.Algebra.Ring.Basic
+import Mathlib.Topology.Algebra.OpenSubgroup
+
+/-! # Linear topologies on rings
+
+Following Bourbaki, *Algebra II*, chapter 4, §2, n° 3, a topology on a ring `R` is *linear* if
+it is invariant by translation and admits a basis of neighborhoods of 0 consisting of
+two-sided ideals.
+
+- `IsLinearTopology.tendsto_mul_zero_of_left`: for `f, g : ι → R` such that `f i` converges to `0`,
+`f i * g i` converges to `0`.
+
+- `IsLinearTopology.tendsto_mul_zero_of_right`: for `f, g : ι → R` such that `g i` converges to `0`,
+`f i * g i` converges to `0`.
+
+-/
+
+open scoped Topology
+open Filter
+
+section Ring
+
+variable (α : Type*) [Ring α]
+
+/-- A topology on a ring is linear if its topology is defined by a family of ideals. -/
+class IsLinearTopology [TopologicalSpace α] [TopologicalRing α] where
```

Comment:
Similar to the comment above: is the translation-invariance mentioned above the same as `[TopologicalRing]` here?

Reply:
Yes, replacing `[TopologicalRing α]` by `[ContinuousConstSMul α α]` would give (mathematically) the same definition. However:
- I've essentially proved that in a PR depending on #18437, so I don't think there's much sense redoing the work here instead of waiting for it to arrive.
- Even when we have these tools, I'm not sure we want the typeclass system to spend too much time synthesizing `TopologicalRing` instances. I would prefer that, in concrete cases, we just prove `TopologicalRing` first (using the special criterion), and then `LinearTopology`.

Reply:
Thanks for clarifying!

In that case I completely agree with the design choice to assume `[TopologicalRing]`. But please point this out. E.g. in a `## Todo` that points to your wip.

Reply:
I have made a small modification: `LinearTopology` doesn't assume `TopologicalRing` anymore. The assumption is only added for those lemmas that require it. I have also added a note.